### PR TITLE
Faststream client and callback wire-in

### PR DESCRIPTION
**Pull request by henricook**
TODO: Callback method is wordy, needs genericising.

Some callback types not accounted for yet as I don't have the interviews setup in Launchpad to create samples (pending admin access and business feed-in)

Requires: https://github.com/hmrc/fset-faststream/pull/146

### CODE REVIEW

**Comment by ollyw** [comment]
Let's hold off on this for this current prod-rel, as it does have callbacks to the fast stream service

**Comment by henricook** [comment]
Yes agree, the faststream side isn't ready yet.

I'm suppressing method length and cyclomatic complexity - i can make it more specific (but will really just genericise it when i'm not rushing for a prototype)

File: app/uk/gov/hmrc/fsetlaunchpadgateway/controllers/CallbackController.scala [hunk 1]
```diff
@@ -29,36 +34,45 @@ trait CallbackController extends FrontendController {
...
           Logger.debug("finished callback received!")
-          Future.successful(Ok("Received"))
+          faststreamClient.finishedCallback(FinishedCallbackRequest.fromExchange(parsed)).map(_ => Ok("Received"))
         case _ =>
           Logger.warn(s"Unknown callback type received! Status was $status, JSON body was $contentAsJson")
           Future.successful(BadRequest("Status was not recognised"))
       }) match {
         case Success(result) => result
-        case Failure(ex) => Future.successful(BadRequest("Request was malformed"))
+        case Failure(ex) if ex.isInstanceOf[JsResultException] =>
```
Comment (ollyw) [inline app/uk/gov/hmrc/fsetlaunchpadgateway/controllers/CallbackController.scala]: Ok, what am I missing? What is wrong with the classic `case Failure(ex: JsResultException(_)) =>` to make it more ~~type safe~~ lightbend?

File: app/uk/gov/hmrc/fsetlaunchpadgateway/controllers/CallbackController.scala [hunk 2]
```diff
@@ -29,36 +34,45 @@ trait CallbackController extends FrontendController {
...
+          Logger.warn(s"Could not parse payload with valid status. Raw request: ${request.body}. Exception: $ex")
+          Future.successful(BadRequest("Request was malformed"))
+        case Failure(ex) if ex.isInstanceOf[CallbackException] =>
+          Logger.warn(s"Error when passing callback to faststream. Raw request: ${request.body}. Exception: $ex")
+          Future.successful(InternalServerError("Request to upstream server failed"))
+        case Failure(ex) =>
+          Logger.warn(s"Could not parse payload. Raw request: ${request.body}")
```
Comment (ollyw) [inline app/uk/gov/hmrc/fsetlaunchpadgateway/controllers/CallbackController.scala]: Done this contain email addresses? Probably not, but just thought I would double check

File: app/uk/gov/hmrc/fsetlaunchpadgateway/controllers/CallbackController.scala (hunk 1 above)
Comment (henricook) [inline app/uk/gov/hmrc/fsetlaunchpadgateway/controllers/CallbackController.scala]: Fixed!

**Comment by henricook** [comment]
Changes pushed @ollyw

**Comment by henricook** [comment]
@ollyw Are you happy with this?

**Pull request merged**